### ISSUE TITLE
# Neural Network to classify MNIST dataset

### ISSUE THREAD

**Issue by prabhakar23jiaiml009-code** [prabhakar23jiaiml009-code/shallow_deep_learning#1]
# Neural Network to classify MNIST dataset (Complete Code)

import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Flatten
from tensorflow.keras.utils import to_categorical

# Load MNIST dataset
(X_train, y_train), (X_test, y_test) = tf.keras.datasets.mnist.load_data()

# Normalize data
X_train = X_train / 255.0
X_test = X_test / 255.0

# One-hot encode labels
y_train = to_categorical(y_train, 10)
y_test = to_categorical(y_test, 10)

# Build model (Shallow Neural Network)
model = Sequential()
model.add(Flatten(input_shape=(28, 28)))
model.add(Dense(128, activation='relu'))
model.add(Dense(10, activation='softmax'))

# Compile model
model.compile(
    optimizer='adam',
    loss='categorical_crossentropy',
    metrics=['accuracy']
)

# Train model
model.fit(X_train, y_train, epochs=5, batch_size=32)

# Evaluate model
loss, accuracy = model.evaluate(X_test, y_test)
print("Test Accuracy:", accuracy)

# Predict example
import numpy as np
sample = X_test[0].reshape(1, 28, 28)
prediction = model.predict(sample)
print("Predicted Digit:", np.argmax(prediction))
